sdk/servicebus/azure-servicebus/azure/servicebus/_common/utils.py

```diff
@@ -217,18 +226,23 @@ def transform_messages_to_sendable_if_needed(messages):
             return messages
 
 
-def _single_message_from_dict(message, message_type):
-    # type: (Union[ServiceBusMessage, Mapping[str, Any]], Type[ServiceBusMessage]) -> ServiceBusMessage
+def _convert_to_single_service_bus_message(message, message_type):
+    # type: (SingleMessageType, Type[ServiceBusMessage]) -> ServiceBusMessage
+    # pylint: disable=protected-access
+    if isinstance(message, AMQPAnnotatedMessage):
+        message = message_type(
```

Comment:
Rather than rebuild a ServiceBusMessage, can we not simply add the `_to_outgoing_message()` function to `AmqpAnnotatedMessage`?

Reply:
updated the AmqpAnnotatedMessage to have the `_to_outgoing_message` method -- but it needs to take a `message_type` parameter. otherwise `AmqpAnnotatedMessage` would have a dependency on `ServiceBusMessage` which results in a circular dependency issue.

also did some internal refactor to eliminate unnecessary message transformation.